teams/vsp/teams/tools/frontend/cypress-migration-guide.md

```diff
@@ -0,0 +1,258 @@
+  .upload('src/platform/testing/example-upload.png', 'image/jpg')
+```
+
+### Accessibility <a name="accessibility"></a>
+Like most E2E tools, Cypress has its own [axe-core plugin](https://github.com/avanslaars/cypress-axe) to test accessibility.
+
+To add aXe checks to your tests use the custom `axeCheck()` command based off the `checkA11y` [command](https://github.com/avanslaars/cypress-axe#cychecka11y).
+
+#### Nightwatch:
+```javascript
+client.axeCheck('.main')
+```
+
+#### Cypress:
+```javascript
+cy.axeCheck();
+```
+
+### Assertions <a name="assertions"></a>
+
+Cypress uses `chai` to [handle assertions](https://docs.cypress.io/guides/references/assertions.html#BDD-Assertions). Currently in the `nightwatch-assertions` folder, we have 4 custom assertions, which can all be replaced:
+
+#### hasFocusableCount / hasTabbableCount
+Checks the count of focusable and tabbable elements.
+
+These can be converted by making [custom chai assertions](https://docs.cypress.io/guides/references/assertions.html#Adding-New-Assertions).
+
+#### isActiveElement
+Checks if the given element is focused on the page.
+
+```javascript
+// Bdd assertion
+cy.get('input').should('have.focus')
+```
+#### isDisabledElement 
+Checks if the given element is disabled on the page.
+
+```javascript
+// Bdd assertion
+cy.get('button').should('be.disabled')
+```
+
+## Running Tests <a name="running-tests"></a>
+
+Cypress supports Chrome, Edge, Firefox, and a few [others](https://docs.cypress.io/guides/guides/launching-browsers.html#Browsers). You can run tests in headless mode or via the test runner.
+
+### Headless mode <a name="headless-mode"></a>
+To run tests headless, run `yarn cypress run --headless`
+
+### Test Runner <a name="test-runner"></a>
+To run tests using the Cypress [test runner](https://docs.cypress.io/guides/core-concepts/test-runner.html#Overview), run `yarn cypress open`.
+
+With the test runner, you are able to use Cypress's "Open Selector Playground". This allows you to click on elements in the DOM and copy that element's selector to use in your test. This means there's usually no need to manually inspect an element and copy the selector.
+
+The test runner provides the ability to pause tests, and [time travel](https://docs.cypress.io/guides/getting-started/writing-your-first-test.html#Time-travel), which allows you to see snapshots of your tests.
+
+You can add any of the [run options](https://docs.cypress.io/guides/guides/command-line.html#Commands) to the commands above.
+
+## Things to Note <a name="things-to-note"></a>
+
+### Automatic Waiting
+Cypress automatically waits for commands to execute before moving on to the next. This eliminates needing to use the timeout consts in `platform/testing/e2e/timeouts.js`. Cypress queues it's commands instead of running them synchronously, so doing something like [this](https://docs.cypress.io/guides/references/best-practices.html#Assigning-Return-Values) will not work.
```

Comment:
```suggestion
Cypress automatically waits for commands to execute before moving on to the next one. This eliminates the need to use the timeout constants in `platform/testing/e2e/timeouts.js`.

Cypress queues its commands instead of running them synchronously, so doing something like [this](https://docs.cypress.io/guides/references/best-practices.html#Assigning-Return-Values) will not work.
```